src/main/java/com/develocity/api/tests/TestsApiSample.java

```diff
@@ -0,0 +1,174 @@
+
+    private static final List<TestOutcome> UNSTABLE_OUTCOMES = Arrays.asList(TestOutcome.FAILED, TestOutcome.FLAKY);
+    private static final List<TestIncludeFields> INCLUDE_BUILD_SCAN_IDS_AND_WORK_UNITS = Arrays.asList(TestIncludeFields.BUILD_SCAN_IDS, TestIncludeFields.WORK_UNITS);
+    private static final Comparator<TestOrContainer> UNSTABLE_TEST_COMPARATOR = Comparator.<TestOrContainer>comparingDouble(testOrContainer -> {
+            TestOutcomeDistribution outcomeDistribution = testOrContainer.getOutcomeDistribution();
+            return (double) (outcomeDistribution.getFailed() + outcomeDistribution.getFlaky()) / outcomeDistribution.getTotal();
+        })
+        .reversed()
+        .thenComparing(TestOrContainer::getName);
+
+    @Mixin
+    GradleEnterpriseApiProvider apiProvider;
+
+    @Option(
+        names = "--project-name",
+        description = "The name of the project to show the containers of (if omitted, containers from all builds are shown)",
+        defaultValue = Option.NULL_VALUE,
+        order = 2
+    )
+    String projectName;
+
+    @Option(
+        names = "--reporter-type",
+        description = "The type of the reporter to use (if omitted, the report will be printed to the standard output)",
+        defaultValue = "STANDARD_OUTPUT",
+        order = 2
+    )
+    ReporterType reporterType;
+
+    @Option(
+        names = "--github-repo",
+        description = "The URL of the GitHub repository to create issues in, required if reporter type is GITHUB_CLI.",
+        order = 3
+    )
+    @Nullable
+    String githubRepoUrl;
+
+    public static void main(String[] args) {
+        System.exit(new CommandLine(new TestsApiSample()).execute(args));
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        GradleEnterpriseApi api = apiProvider.create();
+
+        // builds query does not support a more fine-grained resolution
+        OffsetDateTime now = OffsetDateTime.now().truncatedTo(ChronoUnit.SECONDS);
+
+        Set<String> unstableTestContainersFromLastWeek = getUnstableTestContainersFromLastWeek(api, now);
+        List<TestOrContainer> newUnstableTestContainers = getNewUnstableTestContainers(api, unstableTestContainersFromLastWeek, now);
+        List<TestContainerWithCases> unstableTestContainersWithCases = getUnstableTestCases(api, newUnstableTestContainers, now);
+
+        switch (reporterType) {
+            case STANDARD_OUTPUT:
+                new StandardOutputReporter(apiProvider.getServerUrl(), now, unstableTestContainersWithCases).report();
+                break;
+            case GITHUB_CLI:
+                new GitHubCliReporter(
+                    apiProvider.getServerUrl(),
+                    requireNonNull(githubRepoUrl, "GitHub URL is missing"),
+                    now,
+                    unstableTestContainersWithCases,
+                    new Interval(now.minusDays(1), now)
+                ).report();
+                break;
+        }
```

Comment:
Since we're on Java 8 here, I'd add a default with exception just for safety purposes

Reply:
Done in https://github.com/gradle/gradle-enterprise-api-samples/pull/90/commits/2bc2cebef9edb425a1ccec7b5f050a74e9381b76